python/lsst/pex/config/config.py

```diff
@@ -1382,6 +1404,43 @@ def compare(self, other, shortcut=True, rtol=1E-8, atol=1E-8, output=None):
         return compareConfigs(name, self, other, shortcut=shortcut,
                               rtol=rtol, atol=atol, output=output)
 
+    @classmethod
+    def __init_subclass__(cls, **kwargs):
+        """Run initialization for every subclass.
+
+        Specifically registers the subclass with a YAML representer
+        and YAML constructor (if pyyaml is available)
+        """
+        super().__init_subclass__(**kwargs)
+
+        if not yaml:
+            return
+
+        def config_representer(dumper, data):
```

Comment:
Since this and its paired constructor don't vary by subclass, can they move outside the class?